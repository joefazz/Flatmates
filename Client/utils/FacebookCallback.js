(error, result) => {
    if (error) {
      alert('Error fetching data: ' + error.toString());
    } else {
      alert('Success fetching data');
      console.log(result)
    }
}