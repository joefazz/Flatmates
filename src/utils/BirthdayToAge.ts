import * as moment from 'moment';

export function ConvertBirthdayToAge(birthday: string): number {
    return moment().diff(moment(birthday, 'MM/DD/YYYY'), 'years');
}
