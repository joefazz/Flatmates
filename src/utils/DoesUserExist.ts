import client from '../Client';
import { USER_LOGIN_QUERY } from '../graphql/queries';
import { UserLoginQuery } from '../graphql/Types';
