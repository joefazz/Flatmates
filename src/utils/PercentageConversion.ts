import { Metrics } from '../consts';

enum TYPES {
    Width = 'width',
    Height = 'height',
    Font = 'fontSize'
};

export function toConstantWidth(percentage: number): number {
    return toConstant(percentage, TYPES.Width);
}

export function toConstantHeight(percentage: number): number {
    return toConstant(percentage, TYPES.Height);
}

export function toConstantFontSize(percentage: number): number {
    return toConstant(percentage, TYPES.Font);
}

export function toPercentageHeight(constant: number): number {
    return toPercentage(constant, TYPES.Height);
}

export function toPercentageWidth(constant: number): number {
    return toPercentage(constant, TYPES.Width);
}

export function toPercentageFontSize(constant: number): number {
    return toPercentage(constant, TYPES.Font);
}

function toPercentage(constant: number, type: TYPES): number {
    switch (type) {
        case TYPES.Width:
            return (constant * 100) / Metrics.screenWidth;

        case TYPES.Height:
            return (constant * 100) / Metrics.screenHeight;

        case TYPES.Font:
            return (constant * 100) / Metrics.totalSize;
    }
}

function toConstant(percentage: number, type: TYPES): number {
    switch (type) {
        case TYPES.Width:
            return Metrics.screenWidth * (percentage / 100);
        case TYPES.Height:
            return Metrics.screenHeight * (percentage / 100);
        case TYPES.Font:
            return Metrics.totalSize * (percentage / 100);
    }

    return 0;
}