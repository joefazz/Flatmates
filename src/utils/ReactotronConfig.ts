import Reactotron from 'reactotron-react-native';
import { reactotronRedux } from 'reactotron-redux';
import sagaPlugin from 'reactotron-redux-saga';

var Tron: any = 0;

// if (__DEV__) {
//     Tron = Reactotron.configure()
//         .useReactNative()
//         .use(reactotronRedux())
//         .use(sagaPlugin())
//         .connect();
// }

export default Tron;
