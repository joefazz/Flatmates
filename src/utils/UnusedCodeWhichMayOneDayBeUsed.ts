/* CODE WHICH ANIMATES AN EXPANDABLE FILTER, MADE FOR POST LIST DISCARDED */

// private _animationValue: Animated.Value = new Animated.Value(0);
// private _ANIMATION_DURATION_CONSTANT: number = 500;

// private heightAnimation = {
//     height: this._animationValue.interpolate({
//         inputRange: [0, 1],
//         outputRange: [toConstantHeight(10.5), toConstantHeight(22)]
//     })
// };

// private shrinkGrowAnimation = {
//     width: this._animationValue.interpolate({
//         inputRange: [0, 1],
//         outputRange: [toConstantWidth(27), toConstantWidth(90)]
//     }),
//     // backgroundColor: 'red'
// }

// private opacityAnimation = {
//     opacity: this._animationValue.interpolate({
//         inputRange: [0, 1],
//         outputRange: [0, 1]
//     })
// };

// private rotateAnimation = {
//     transform: [
//         {
//             rotate: this._animationValue.interpolate({
//                 inputRange: [0, 1],
//                 outputRange: ['0deg', '180deg']
//             })
//         }
//     ]
// }

{
    /*<TouchableOpacity onPress={this.animateFilter} activeOpacity={0.7} style={ feed.expandBar }>
                        <Animated.View style={this.rotateAnimation}>
                            <Icon name={'ios-arrow-up'} size={toConstantFontSize(3)} style={{color: Colors.highlightWhite}} />
                        </Animated.View>
                    </TouchableOpacity>*/
}

// private animateFilter = () => {
//     if (this.state.isFilterOpen) {
//         Animated.timing(this._animationValue, {
//             toValue: 0,
//             duration: this._ANIMATION_DURATION_CONSTANT,
//             easing: Easing.elastic(0.7)
//         }).start(() => this.setState({ isFilterOpen: false }));
//     } else {
//         Animated.timing(this._animationValue, {
//             toValue: 1,
//             duration: this._ANIMATION_DURATION_CONSTANT,
//             easing: Easing.elastic(1)
//         }).start(() => this.setState({ isFilterOpen: true }));
//     }
// }

/* ---------------------------------------------- */
