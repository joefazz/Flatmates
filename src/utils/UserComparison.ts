// @ts-ignore
import moment from 'moment';

export function compareUsers(me, otherUser): number {
    // smoker - 0.3
    // gender - 0.3
    // course -  - 0.2
    // study year - 0.1
    // birthday - 0.05
    // facebook info - 0.05
    let score = 0;

    if (me.isSmoker === otherUser.isSmoker) {
        score += 0.3;
    }

    // Add gender preference comparision
    if (me.gender === otherUser.gender) {
        score += 0.3;
    }

    if (me.course === otherUser.course) {
        score += 0.2;
    }

    if (me.studyYear === otherUser.studyYear) {
        score += 0.1;
    }

    if (moment(me.birthday).format('YYYY') === moment(otherUser.birthday).format('YYYY')) {
        score += 0.05
    }

    // leave facebook

    return Math.floor(score * 100);
}
