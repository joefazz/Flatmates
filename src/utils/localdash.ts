import { MapboxSDK } from '../App';

function capitalize(text: string): string {
    return text.charAt(0).toUpperCase() + text.slice(1);
}

async function getCoordsFromAddress(road: string): Promise<string | Array<number>> {
    const address: string = road + ', Reading';
    let res: any;

    try {
        res = await MapboxSDK.geocodeForward(address, {
            country: 'gb',
            proximity: { latitude: 51.4412, longitude: -0.943 }
        });
    } catch (err) {
        alert(
            'There was a problem with the road name you entered, please check it and try again'
        );
        return 'Error';
    }

    if (res.status !== 200 || res.entity.features.length === 0) {
        alert(
            'There was a problem with the road name you entered, please check it and try again.'
        );
        return 'Error';
    } else {
        return res.entity.features[0].center;
    }
}

export { capitalize, getCoordsFromAddress };