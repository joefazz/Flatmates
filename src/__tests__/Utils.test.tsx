import { ConvertBirthdayToAge } from '../utils/BirthdayToAge';

test('Convert a birthday string to an age', () => {
    let birthday = '10/21/1996';

    expect(ConvertBirthdayToAge(birthday)).toMatchSnapshot();
})