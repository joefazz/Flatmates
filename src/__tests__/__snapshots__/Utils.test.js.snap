// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert a birthday string to an age 1`] = `21`;
